Mathlib/Data/Multiset/Basic.lean

```diff
@@ -2327,6 +2332,57 @@ theorem map_count_True_eq_filter_card (s : Multiset α) (p : α → Prop) [Decid
   simp only [count_eq_card_filter_eq, filter_map, card_map, Function.id_comp,
     eq_true_eq_id, Function.comp_apply]
 
+theorem sub_singleton [DecidableEq α] (a : α) (s : Multiset α) : s - {a} = s.erase a := by
+  ext
+  simp [Multiset.erase_singleton, Multiset.count_singleton]
+  split <;> simp_all
+
+theorem singleton_add_sub_of_cons_add [DecidableEq α] {a a0: α} {M X : Multiset α}
+    (H : a ::ₘ M = X + {a0}) : M = X + {a0} - {a} := by
+  by_cases hyp : a = a0
+  · rw [hyp, add_comm] at H
+    simp_all [Multiset.singleton_add]
+  · have a0_a: a0 ≠ a := by rw [eq_comm] at hyp; exact hyp
+    ext b
+    simp [Multiset.count_cons, Multiset.count_singleton, Multiset.count_add]
+    have H : Multiset.count b (a ::ₘ M) = Multiset.count b (X + {a0}) := by simp_all only
+    [Multiset.count_add]
+    by_cases ba : b = a
+    · rw [ba] at *
+      have : (a ::ₘ M).count a = M.count a + 1 := by simp
+      simp_all
+    by_cases ba0 : b = a0
+    · have : (a ::ₘ M).count a0 = X.count a0 + 1 := by
+        subst_eqs
+        rw [add_comm, Multiset.singleton_add] at H
+        simp_all
+      have : M.count a0 = Multiset.count a0 (a ::ₘ M) := by
+        have : a0 ≠ a := by simp_all
+        rw [Multiset.count_cons_of_ne this M]
+      simp_all
+    · have : M.count b = (a ::ₘ M).count b := by
+        have : b ≠ a := by simp_all
+        rw [Multiset.count_cons_of_ne this M]
+      rw [this]
+      simp_all
+
+theorem mem_sub_of_not_mem_of_mem {α} [DecidableEq α] (x : α) (X Y: Multiset α) (x_in_X : x ∈ X)
+    (x_notin_Y : x ∉ Y) : x ∈ X - Y  := by
```

Comment:
```suggestion
theorem mem_sub [DecidableEq α] : a ∈ s - t ↔ t.count a < s.count a := by
  rw [← count_pos, count_sub, Nat.sub_pos_iff_lt]
```
then your original lemma can be obtained using `count_pos` again and `count_eq_zero_of_not_mem`

Reply:
Indeed, (also as a side note to myself ;-)) the original lemma can be proved as follows:

`
theorem mem_sub_of_not_mem_of_mem {α} [DecidableEq α] (x : α) (X Y: Multiset α) (x_in_X : x ∈ X)
    (x_notin_Y : x ∉ Y) : x ∈ X - Y := by
  rwa [mem_sub, count_eq_zero_of_not_mem, count_pos]
  assumption
`